### PR TITLE
Implementation of use_k parameters in descriptors and forward map.

### PR DESCRIPTION
**Pull request by gviga**
This Pr partially handles #79 in learning module introducing some parameters to better handling use_k of basis.

Moreover it solves a bug about doctrings and dimension of functional map in leanring models and improved the loss file

### CODE REVIEW

File: geomfum/descriptor/_base.py [hunk 1]
```diff
@@ -44,20 +55,26 @@ def __call__(self, shape):
         shape : Shape.
             Shape.
         """
+        if self.k is not None:
+            if shape.basis.spectrum_size != self.k:
+                shape.basis.use_k = self.k
```
Comment (GiLonga) [inline geomfum/descriptor/_base.py:60]: Maybe would be better you utilize the setter function in the class basis

**Comment by GiLonga** [comment]
This PR is mergeable, but I don't think it resolves #79, but almost it is a fix for the descriptors. We should discuss a new design for the use_k parameter separately.

**Review by GiLonga — APPROVED** [review]

File: geomfum/forward_functional_map.py [hunk 2]
```diff
@@ -20,14 +20,17 @@ class ForwardFunctionalMap(abc.ABC, nn.Module):
         Resolvant of the regularized functional map (default: 1).
     bijective: bool
         Whether we compute the map in both the directions (default: True).
+    fmap_shape: tuple, optional
+        Shape of fmap12, i.e (spectrum_size_b, spectrum_size_a). If None, the shape is inferred from the input shapes.
```
[the remark this answers is not in the document]
Comment (gviga) [reply geomfum/forward_functional_map.py:24]: If none, the shape of fmap Will be basis_b.use_k, basis_a.use_k

File: geomfum/descriptor/_base.py (hunk 1 above)
Comment (GiLonga) [reply geomfum/descriptor/_base.py:60]: Totally, my bad, sorry
